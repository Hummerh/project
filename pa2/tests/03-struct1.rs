struct Point {x : i32, y : i32, z : i32}

struct Dog {
    color: i32,
    model: i32,
    width: i32,
    tailless: bool
}
