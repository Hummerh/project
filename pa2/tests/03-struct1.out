(crate
    (items
        (struct-def
            (id
                (Point))
            (field-defs
                (field-def
                    (id
                        (x))
                    (type-i32))
                (field-def
                    (id
                        (y))
                    (type-i32))
                (field-def
                    (id
                        (z))
                    (type-i32))))
        (struct-def
            (id
                (Dog))
            (field-defs
                (field-def
                    (id
                        (color))
                    (type-i32))
                (field-def
                    (id
                        (model))
                    (type-i32))
                (field-def
                    (id
                        (width))
                    (type-i32))
                (field-def
                    (id
                        (tailless))
                    (type-bool))))))